source/core_mqtt.c

```diff
@@ -629,8 +628,7 @@ static bool matchWildcards( const char * pTopicName,
 
         /* If the topic name contains a child level but the topic filter ends at
          * the current level, then there does not exist a match. */
-        if( ( nextLevelExistsInTopicName == true ) &&
-            ( nextLevelExistsinTopicFilter == false ) )
+        if( nextLevelExistsInTopicName && !nextLevelExistsinTopicFilter )
```

Comment:
I would prefer the original to the modified version. The original seems more clear to me.